scripts/tls_test.c

```diff
@@ -208,12 +212,24 @@ static bool load_certificate(char *filename)
 /**
  * Load private key from file
  */
-static bool load_key(char *filename)
+static bool load_key(tls_version_t min_version, tls_version_t max_version,
+					 char *filename)
 {
 	private_key_t *key;
+	enumerator_t *enumerator;
+	key_type_t type;
 
-	key = lib->creds->create(lib->creds, CRED_PRIVATE_KEY, KEY_RSA,
-							  BUILD_FROM_FILE, filename, BUILD_END);
+	enumerator = tls_get_supported_key_types(min_version, max_version);
+	while (enumerator->enumerate(enumerator, &type))
+	{
+		key = lib->creds->create(lib->creds, CRED_PRIVATE_KEY, type,
+								 BUILD_FROM_FILE, filename, BUILD_END);
+		if (key)
+		{
+			break;
+		}
+	}
+	enumerator->destroy(enumerator);
```

Comment:
This approach works fine although it does not seem to be used that way.

`lib->freds->create()` throws failures:
```
building CRED_PRIVATE_KEY - ECDSA failed, tried 3 builders
building CRED_PRIVATE_KEY - ED25519 failed, tried 3 builders
building CRED_PRIVATE_KEY - ED448 failed, tried 3 builders
```

Reply:
Yeah, I'd just load the key with `KEY_ANY` as only one key will be loaded anyway and auto-detection should generally work (depends a bit on the loaded plugins).

Reply:
Yes I agree. I tried to use `key = lib->creds->create(lib->creds, CRED_PRIVATE_KEY, KEY_ANY, BUILD_FROM_FILE, filename, BUILD_END);` but `construct` is not initialised by `construct = entry->constructor(subtype, args);` in `credential_factory-c` when trying with a subtype 0 (`ANY`). 

When creating a `private_key_t` object with `KEY_ED25519` it is possible to load the ED25519 key though. I interpret the plugins are loaded correctly but the key auto-detection seems to miss EDDSA keys in combination with `KEY_ANY`.

Reply:
I guess that depends on the loaded plugins and the key format. For instance, the _openssl_ plugin can load an Ed25519 key even with `KEY_RSA` (for private keys, the plugin only does auto-detection and ignores the given type). The same applies if the key is in PKCS#8 format and the _pkcs8_ plugin is loaded, which should pass along the correct key type (indicated by the algorithm identifier in the structure) for plugins that don't do auto-detection. What plugins have you loaded? What format does your key have?

Reply:
You probably need the _pkcs8_ plugin.